### DISCUSSION TITLE
Any suggestion for making dev env faster

### DISCUSSION THREAD

**Discussion by caibirdme** [vectordotdev/vector#14555] [Q&A] [answered]
vector is such a huge project that everytime I change some lines of code, I have to wait for rust-analyzer running `cargo check` for over 2 minutes. If I run some tests, it'll trigger linking, and it's even cost me over 5~8 minutes to wait for the results. 
That's really frustrating for me to contribute to this project. Any idea to make compiling or developing a little bit happier? I don't know how your dev experience look like

**Comment by jszwedko** [comment, marked as answer]
Hi @caibirdme !

You can see some tips here for speeding up compilation: https://github.com/vectordotdev/vector/blob/master/docs/DEVELOPING.md#tips-and-tricks . The most effective is using feature flags to only build the part of Vector you are actively developing (e.g. `cargo check (or build or test) --no-default-features --features "sinks-elasticsearch"` if you are just modifying the Elasticsearch sink). You can see the full list of feature flags in `Cargo.toml`.

We are hoping to make some improvements here in the future to make it easier to iterate on sub-components of Vector without needing to do a full build.
